References updates (Issue #1818)

SpawnProd opened the pull request:
Updates for Rx, ReactiveUI and Microsoft CSharp nuget references.

Fixes #1818

SpawnProd committed:
References updates

Updates for Rx, ReactiveUI and Microsoft CSharp nuget references.

jmacato reviewed (changes requested):
```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(1590,2): error : Exception while loading assemblies: System.IO.FileNotFoundException: Could not load assembly 'System.Threading.Tasks.Extensions, Version=4.2.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. Perhaps it doesn't exist in the Mono for Android profile? 
```
CI Failed with that, seems like there's some conflict with Xamarin perhaps?

SpawnProd committed:
Downgrade to Rx 4.0

Because of MonoAndroid we can't use Rx 4.1 (System.Threading.Tasks.Extensions is duplicated in mscorlib)

jmacato reviewed (approved):
LGTM!
